src/EditMemoPanel.js

```diff
@@ -0,0 +1,56 @@
+export default function EditMemoPanel({
+  updateIsEditing,
+  memoInEdit,
+  updateMemoInEdit,
+  saveMemoInEdit,
+  deleteMemoInEdit,
+}) {
+  function clearMemoInEdit() {
+    updateMemoInEdit({ id: null, content: "" });
+  }
+
+  function handleEditButtonClick() {
+    if (!memoInEdit.content) {
+      return;
+    }
+
+    saveMemoInEdit();
+    updateIsEditing(false);
+    clearMemoInEdit();
+  }
+
+  function handleDeleteButtonClick() {
+    deleteMemoInEdit();
+    updateIsEditing(false);
+  }
+
+  function onMemoTextAreaChange(e) {
+    updateMemoInEdit({ id: memoInEdit.id || null, content: e.target.value });
+  }
+
+  return (
+    <div className="section">
+      <textarea
+        value={memoInEdit.content || ""}
+        onChange={onMemoTextAreaChange}
+      />
+      <button
+        className="add-button"
+        type="submit"
+        onClick={handleEditButtonClick}
+      >
+        保存
+      </button>
+      <button
+        className="delete-button"
+        type="submit"
+        onClick={handleDeleteButtonClick}
+      >
+        削除
+      </button>
+      {memoInEdit.content === "" ? (
+        <div className="error-message">空のメモは保存できません</div>
+      ) : null}
```

Comment:
ある条件を満たすときだけ要素を表示する、というのは下記のように書くのが通例です。
```suggestion
      {memoInEdit.content === "" && <div className="error-message">空のメモは保存できません</div>}
```

Reply:
ありがとうございます。ご指摘いただいた点修正してみました 🙇‍♂️ 

**localStorage のデータ設計**
987259a24aebad318907228cecd87d5151a3d819

**インクリメントの書き方を修正**
81482e9c2ce103158545bc8102b47222dde82a0d

**条件によって表示を変える処理の書き方を修正**
1e2326c19346941a9f1f592f193c0bb37774efa3